OfflineUX.md

```diff
@@ -0,0 +1,138 @@
+
+6. Is there validation done on the server?
+
+If there is some validation happening on the server that needs to happen before the feature can work, then we answer YES to this question. Remember, this is referring to validation that cannot happen on the front end (e.g. reusing an existing password when resetting a password). For example, if we want to set up a bank account then our answer to the previous question (5) is YES (we could imagine that the server returns a 200 and we would know what the next page would look like). However, our answer to this question is NO (because we can’t suggest to the user that their request succeeded when really it hasn’t been sent yet–their card wouldn’t work!)
+
+This question can be tricky, so if you’re unsure, please ask a question in #expensify-open-source and tag the contributor management engineer team.
+
+7. Does the user need to know if the action was successful?
+
+Think back to the pinning example from above: the user doesn’t need to know that their pinned report's NVP has been updated. To them the impact of clicking the pin button is that their chat is at the top of the LHN. It makes no difference to them if the server has been updated or not, so the answer would be NO. Now let’s consider the case of sending a comment. In this example, the user needs to know if their comment was actually sent, because they may need to know if the person they sent it to can read it. In this case our answer is YES.
+
+### Descriptions of the UX Patterns
+
+# None - No Offline Behavior
+
+ There’s no specific UI for this case. The feature either looks totally normal and works as expected (because it doesn’t need the server to function) or the feature looks like it did whenever connection was lost. 
+
+Used when…
+ - there is no interaction with the server in any way
+ - or data is READ from the server and does not need to show up- to- date data. The user will see stale data until the new data is put into Onyx and then the view updates to show the new data. 
+
+# A - Optimistic Without Feedback Pattern
+
+This is the pattern where we queue the request to be sent when the user is online and we continue as if the request succeeded. 
+
+Used when…
+ - the user should be given instant feedback and
+ - there is no error handling in the server and
+ - the user does not need to know when the change is done on the server in the background
+
+How to implement: Use API.queue() to implement this pattern. 
+
+# B - Optimistic WITH Feedback Pattern
+This pattern queues the API request, but also makes sure that the user is aware that the request hasn’t been sent yet.
```

Comment:
>makes sure that the user is aware that the request hasn’t been sent yet

Please make it clear that this ONLY should happen when the user is offline. If the user is online it should just look like it has succeeded.